doc/contributing/docs/terms.rst

```diff
@@ -0,0 +1,119 @@
+Using Terminology
+=================
+
+Concepts and Terms
+------------------
+
+To write well about some subject matter, one needs to know the objects of this subject matter
+and use the right, carefully selected words for them.
+Such objects are called concepts, and the words for them are called terms.
+
+..  glossary::
+
+    concept
+        Concept is the the idea of some object, attribute or action.
+        It is independent of languages, audience and products, it just exists.
+
+        For example, a large database can be partitioned into smaller instances,
+        which are easier to operate and together can exceed the throughput of
+        a single large database instance. Such instances can exchange data
+        to keep it consistent between them.
+
+    term
+        Term is a word, which authors or a particular book, article or documentation set
+        have explicitly selected to denote a :term:`concept` in a particular language
+        and for a particular audience.
+
+        For example, in Tarantool we use the term "[database] sharding" to denote the type of database
+        partitioning mentioned earlier.
+
+Use Preferred Terms
+-------------------
+
+The purpose of using terms is to write concisely and unambiguously,
+which is good for readers.
+But selecting terms is hard.
+Often there are two or more terms for one concept, popular with the community,
+so there's no obvious choice.
+Indeed, selecting and consistently using any of them is much better,
+than not selecting and just using random term each time.
```

Comment:
Perhaps we can make a more obvious logical bridge here to the next paragraph that defines the `restricted term`. By adding a sentence like this: "That's why it's also helpful to explicitly restrict the usage of some terms."